### PR TITLE
Refactoring dump read functions to use io.Reader rather than paths

### PR DESCRIPTION
**Pull request by denisonbarbosa**
Also adjusted integration_test.go to properly interact with the changes.

### CODE REVIEW

File: pam/integration_test.go [hunk 1]
```diff
@@ -138,16 +138,16 @@ func TestPamSmAuthenticate(t *testing.T) {
 				ref := filepath.Join(cacheDir, db)
 				wants[db] = testutils.LoadAndUpdateFromGoldenDump(t, ref)
 
-				// TODO: make this better, using a reader
 				// Load temporary got to memory
-				tmp := filepath.Join(t.TempDir(), db+".dump")
-				f, err := os.Create(tmp)
+				f, err := os.Create(filepath.Join(t.TempDir(), db+".dump"))
```
Comment (didrocks) [inline pam/integration_test.go:142]: didn’t we say we would use an in memory buffer which can write an read from?

**Review by didrocks — CHANGES_REQUESTED** [review]
changes in `db.go` are perfect.

However, in `integration_test.go`, the whole goal is to use a in memory buffer rather than a file, mind changing it to this?

File: pam/integration_test.go (hunk 1 above)
Comment (denisonbarbosa) [reply pam/integration_test.go:142]: Yeah, we did. Probably was a brain fry moment. My bad... 
(In my defense, we did say a lot of stuff yesterday 🥲).

**Review by didrocks — APPROVED** [review]
Thanks for the changes! Let’s merge it :)

**Pull request merged**